base/compiler/typeinfer.jl

```diff
@@ -51,9 +51,10 @@ struct Timing
     cur_start_time::UInt64
     time::UInt64
     children::Core.Array{Timing,1}
-    Timing(mi_info, start_time, cur_start_time, time, children) = new(mi_info, start_time, cur_start_time, time, children)
-    Timing(mi_info, start_time) = new(mi_info, start_time, start_time, UInt64(0), Timing[])
+    bt         # backtrace collected upon initial entry to typeinf
 end
+Timing(mi_info, start_time, cur_start_time, time, children) = Timing(mi_info, start_time, cur_start_time, time, children, nothing)
+Timing(mi_info, start_time) = Timing(mi_info, start_time, start_time, UInt64(0), Timing[])
```

Comment:
I think you can just delete the first one here, actually! :) I think I only had it there because i was forced to define both since i was using inner constructors.

The outer constructor is nicer, and means you can just use the auto-generated default constructor here! :)

```suggestion
Timing(mi_info, start_time) = Timing(mi_info, start_time, start_time, UInt64(0), Timing[], nothing)
```

👍

Reply:
The first one now supplies `nothing` for `bt` and then calls the default constructor.

Reply:
Ah, yeah but the first constructor is now completely unused (other than in your second constructor) because we're now using the default, julia-defined constructor on e.g. line 102 and elsewhere.

So i think it'd be better to delete if possible :)